lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -0,0 +1,243 @@
+      self.default_currency = 'USD'
+      self.supported_cardtypes = %i[visa master american_express discover]
+
+      self.homepage_url = 'https://www.rapyd.net/'
+      self.display_name = 'Rapyd Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      def initialize(options = {})
+        requires!(options, :secret_key, :access_key)
+        super
+      end
+
+      def purchase(money, payment, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+        post[:capture] = true if payment_is_card?(options)
+
+        if payment_is_ach?(options)
+          MultiResponse.run do |r|
+            r.process { commit(:post, 'payments', post) }
+            post = {}
+            post[:token] = r.authorization
+            post[:param2] = r.params.dig('data', 'original_amount').to_s
+            r.process { commit(:post, 'payments/completePayment', post) }
+          end
+        else
+          commit(:post, 'payments', post)
+        end
+      end
+
+      def authorize(money, payment, options = {})
+        post = {}
+        add_invoice(post, money, options)
+        add_payment(post, payment, options)
+        add_address(post, payment, options)
+        post[:capture] = false
+        commit(:post, 'payments', post)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        commit(:post, "payments/#{authorization}/capture", post)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        post[:payment] = authorization
+        add_invoice(post, money, options)
+        commit(:post, 'refunds', post)
+      end
+
+      def void(authorization, options = {})
+        post = {}
+        commit(:delete, "payments/#{authorization}", post)
+      end
+
+      # Gateway returns an error if trying to run a $0 auth as invalid payment amount
+      # Gateway does not support void on a card transaction and refunds can only be done on completed transactions (such as a purchase). Authorize transactions are considered 'active' and not 'complete'  until they are captured.
+      def verify(credit_card, options = {})
+        MultiResponse.run do |r|
+          r.process { purchase(100, credit_card, options) }
+          r.process { refund(100, r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Access_key: )\w+), '\1[FILTERED]').
+          gsub(%r(("number\\?":\\?")\d+), '\1[FILTERED]').
+          gsub(%r(("cvv\\?":\\?")\d+), '\1[FILTERED]')
+      end
+
+      private
+
+      def payment_is_ach?(options)
+        return true if options[:type].include?('bank')
+      end
+
+      def payment_is_card?(options)
+        return true if options[:type].include?('card')
```

Comment:
I love these little methods. I have two questions about the logic here.

1) Are these the only two options? Or could somebody pass a valid `type` that does not include `'bank'` or `'card'`?

2) Is there a world where something like `mx_nationalbank_card` is a valid `type`?